frontend/views/containers/chatroom/ChatMain.vue

```diff
@@ -998,11 +998,13 @@ export default ({
               if (!fromOurselves && isScrollable) {
                 this.updateScroll()
               } else {
-                const msg = this.messages[this.messages.length - 1]
-                this.updateReadUntilMessageHash({
-                  messageHash: msg.hash,
-                  createdHeight: msg.height
-                })
+                const msg = this.messages.filter(m => !m.pending).slice(-1)
```

Comment:
I think this returns array.
```
const msg = this.messages.filter(m => !m.pending).slice(-1)[0]

```

And we don't need to filter here `.filter(m => !m.pending)` because the last message can't be pending here.
```
const msg = this.messages[this.messages.length - 1]
```

Reply:
> I think this returns array.
Yes, good catch. I should have used  `.pop()` instead.

>  because the last message can't be pending here.
Maybe, I need to double check that. But if you're right, it can be simplified to what it was.

Reply:
> Yes, good catch. I should have used .pop() instead.

`.pop()` can't be solution too.